pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -346,6 +347,139 @@ However, as blocks are currently executed sequentially, the
 `Building executable` block should finish in 5 hours minus the time it took the
 `Creating Docker Image` block to finish.
 
+## fail_fast
+
+The `fail_fast` property enables you to set a desired policy for behavior of
+other blocks when one of them fails.
+
+If all blocks are run sequentially, this is not necessary because if one block
+fails, the ones in a sequence after it won't be started and whole pipeline will
+fail immediately.
+
+However, if you have blocks in your pipeline running in parallel (this can be
+achieved by setting [dependencies](#dependencies-in-blocks) property for each
+block), when one block fails, others which do not directly depend on it will
+continue to execute and use time and resources even though whole pipeline will
+eventually fail.
+
+To optimize this, you can choose to set either `cancel` or `stop` fail_fast
+policy.
+
+If `cancel` policy is set, when one block fails the blocks which have not
+yet started executing will be canceled, but already running blocks will be
+allowed to finish executing. This will provide you with more data for debugging
+without occupying any more resources than what was occupied in a first place.
+
+If `stop` policy is set, all running blocks will be stopped and all pending ones
+will be canceled as soon as possible when one block fails.
+
+The `fail_fast` property can have two sub properties, `stop` and `cancel`.
+
+At least one of them is required and if both are stated, `stop` will be
+evaluated first.
+
+Both `stop` and `cancel` properties have only one required property, and that is
+`when` property which holds condition written in
+[Conditions DSL][conditions-reference].
+
+If this condition is fulfilled for given pipeline execution, appropriate
```

Comment:
by "given pipeline execution" you mean workflow?

Reply:
I would rather use pipeline here. 

The conditions are evaluated for a particular pipeline. 

Although all attributes (ie. branch and tag) currently used in conditions are also same for each pipeline in whole workflow, that probably won't always be the case in the future (ie. execution result, values of environment variables etc.).

This is why I think we should stimulate people to always consider conditions in proper context, which is in this case a pipeline.

Reply:
👍